cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/ResponseUtils.java

```diff
@@ -47,16 +51,27 @@ private static void writeResponseToOutputStream(HttpServletResponse response, in
     out.flush();
   }
 
-  public static void returnProgress(HttpServletResponse response, OperationFuture future, boolean json) throws IOException {
+  public static void returnProgress(HttpServletResponse response, List<OperationFuture> futures, boolean json) throws IOException {
     String responseMsg;
     if (json) {
       Map<String, Object> respMap = new HashMap<>();
       respMap.put(VERSION, JSON_VERSION);
-      respMap.put(PROGRESS, future.getJsonArray());
+      List<Object> progress = new ArrayList<>(futures.size());
+      for (OperationFuture future: futures) {
+        Map<String, Object> operationProgress = new HashMap<>(2);
+        operationProgress.put(OPERATION, future.operation());
+        operationProgress.put(OPERATION_PROGRESS, future.getJsonArray());
+        progress.add(operationProgress);
+      }
+      respMap.put(PROGRESS, progress);
       Gson gson = new GsonBuilder().serializeNulls().serializeSpecialFloatingPointValues().create();
       responseMsg = gson.toJson(respMap);
     } else {
-      responseMsg = future.progressString();
+      StringBuilder sb = new StringBuilder();
+      for (OperationFuture operationFuture: futures) {
+        sb.append(String.format("%s:%n%s", operationFuture.operation(), operationFuture.progressString()));
```

Comment:
Is it intentionally `"%s:%n%s"` or should we make the following change:
`"%s:%n%s"` -> `"%s:%s%n"`

Reply:
It is intentional. the intended format is
`operation1_name`
`step1_name`-`progress`
`step2_name`-`progress`
...
`operation2_name`
`step1_name`-`progress`
`step2_name`-`progress`
...